tf5muxserver/mux_server_UpgradeResourceIdentity.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tf5muxserver
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	"github.com/hashicorp/terraform-plugin-mux/internal/logging"
+)
+
+// UpgradeResourceState calls the UpgradeResourceState method, passing `req`,
+// on the provider that returned the resource specified by req.TypeName in its
+// schema.
+func (s *muxServer) UpgradeResourceIdentity(ctx context.Context, req *tfprotov5.UpgradeResourceIdentityRequest) (*tfprotov5.UpgradeResourceIdentityResponse, error) {
+	rpc := "UpgradeResourceIdentity"
+	ctx = logging.InitContext(ctx)
+	ctx = logging.RpcContext(ctx, rpc)
+
+	server, diags, err := s.getIdentityResourceServer(ctx, req.TypeName)
```

Comment:
If we remove the resource identity routing, we can just grab the resource server here

Reply:
done